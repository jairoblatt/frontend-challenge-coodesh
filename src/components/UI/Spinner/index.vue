<template>
  <div class="spinner-container">
    <span class="spinner" :style="styles"></span>
  </div>
</template>
<script lang="ts">
import Vue, { PropOptions } from "vue";
import CSS from "csstype";

export default Vue.extend({
  name: "Spinner",

  props: {
    size: {
      type: [String, Number],
      default: "inherit",
    } as PropOptions<string | number>,
  },

  computed: {
    styles(): CSS.Properties {
      return {
        minHeight: `${this.size}px`,
        minWidth: `${this.size}px`,
      };
    },
  },
});
</script>
<style lang="postcss" scoped>
.spinner-container {
  @apply relative flex justify-center items-center min-h-full min-w-full;
}

.spinner {
  @apply pointer-events-none border-solid rounded-full  border-primary-base  p-1;

  border-width: 0.3rem;
  border-top-color: white;
  animation: spinner 0.5s linear infinite;
}

@keyframes spinner {
  100% {
    transform: rotate(360deg);
  }
}
</style>
