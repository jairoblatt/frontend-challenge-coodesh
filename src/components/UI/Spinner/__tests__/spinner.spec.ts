import { mount, MountOptions, Wrapper } from "@vue/test-utils";
import Spinner from "../index.vue";

describe("Spinner.vue", () => {
  type Instance = InstanceType<typeof Spinner>;
  let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>;

  beforeEach(() => {
    mountFunction = (config = {}) => {
      return mount(Spinner, {
        ...config,
      });
    };
  });

  it("Should mount the componente", () => {
    const wrapper = mountFunction();

    expect(wrapper.vm).toBeDefined();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Should set minHeight and minWidth of element when the "size" prop is passed', () => {
    const wrapper = mountFunction({
      propsData: {
        size: 20,
      },
    });

    const spinner = wrapper.find(".spinner").attributes().style;
    expect(spinner).toBe("min-height: 20px; min-width: 20px;");
  });
});
