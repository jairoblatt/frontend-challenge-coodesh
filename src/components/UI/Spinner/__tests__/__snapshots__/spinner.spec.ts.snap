// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner.vue Should mount the componente 1`] = `"<div class=\\"spinner-container\\"><span class=\\"spinner\\" style=\\"min-height: inheritpx; min-width: inheritpx;\\"></span></div>"`;
